backends/dpdk/dpdkHelpers.cpp

```diff
@@ -765,6 +765,33 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
             } else {
                 ::error("%1%: unhandled function", s);
             }
+        } else if (a->method->name == "mirror_packet") {
+            auto slot_id = a->expr->arguments->at(0)->expression;
+            auto session_id = a->expr->arguments->at(1)->expression;
+            const IR::Expression *slot_idexpr = slot_id;
+            const IR::Expression *session_idexpr = session_id;
+            // mirror instruction supports only metadata as operands, move any constant arguments
+            // to metadata fields
+            if (slot_id->is<IR::Constant>()){
+                IR::ID tmpName(refmap->newName("mirrorSlot"));
+                BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                metadataStruct->fields.push_back(new IR::StructField(tmpName, slot_id->type));
+                auto tmpMember = new IR::Member(new IR::PathExpression("m"), tmpName);
+                add_instr(new IR::DpdkMovStatement(tmpMember, slot_id));
+                slot_idexpr = tmpMember;
+            }
+            if (session_id->is<IR::Constant>()){
+                unsigned value =  session_id->to<IR::Constant>()->asUnsigned();
+                if (value == 0)
+                    ::error("Mirror session ID 0 is reserved for use by Architecture");
+                IR::ID tmpName(refmap->newName("mirrorSession"));
+                BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
```

Comment:
can we have multiple mirror call?
and how the same name parameter adding again will behave?

Reply:
Yes, We can have multiple mirror calls. Compiler would generate 1 mirror instruction for each mirror call. 

If the mirror call is for the same slot id, DPDK target would take care of generating the mirror copy for the last session id.

Regarding the name, refmap newName is used to create the name, so it will take care of creating the names suffixed with incremental IDs.